src/flint/flint_base/flint_base.pyx

```diff
@@ -869,6 +941,87 @@ cdef class flint_mpoly(flint_elem):
         """
         return zip(self.monoms(), self.coeffs())
 
+    def drop_unused_gens(self):
+        """
+        Remove unused generators from this polynomial. Returns a potentially new
+        context, and potentially new polynomial.
```

Comment:
In general I don't like the idea of generating a new context on the fly like this especially based on the value of the polynomial.

I think it should be up to the caller to have the context that they want to use. There could be a context method like `ctx.drop_gens()` that returns a new context and then the caller should use that for coercing to that specific context.

Reply:
Oh, I see there is a `ctx.drop_gens` method.

What is this particular `mpoly.drop_unused_gens` method needed for?

I think I would prefer something like:
```python
ctx2 = ctx.drop_gens(p.unused_gens())
p2 = ctx2(p)
```
In general if you did want to drop some generators you would want to check the generators that are used or not from several polynomials. Perhaps like:
```python
ctx2, [p1_2, p2_2, ...] = ctx.drop_unused_gens([p1, p2, ...])
```

Reply:
> What is this particular mpoly.drop_unused_gens method needed for?

Just a convenience method from before that skipped the index -> gen name -> index process. `unused_gens` is definitely more useful and easier enough to compose.

> In general if you did want to drop some generators you would want to check the generators that are used or not from several polynomials. Perhaps like:
> 
> ```python
> ctx2, [p1_2, p2_2, ...] = ctx.drop_unused_gens([p1, p2, ...])
> ```

Certainly can add this if desired